File: musicpro/views/productosLista.py
from .conn import *
from .moneda import getLocale


def viewProductosLista(request):
    local = getLocale(request)
    productos = Producto.objects.all()
    if "buscar" in request.GET:
        productos = productos.filter(
            nombre__icontains=request.GET.get("buscar"),
            descripcion__icontains=request.GET.get("buscar"),
        )
    return render(
        request, "productosLista.html", {"productos": productos, "local": local}
    )


@csrf_exempt
def newProducto(request):
    if request.method == "POST" and "fil_foto" in request.FILES:
        fs = FileSystemStorage()
        archivo = request.FILES["fil_foto"]
        narchivo = fs.save(archivo.name, archivo)
        producto = Producto.objects.create(
            nombre=request.POST.get("txt_nombre"),
            foto=archivo.name,
            marca=request.POST.get("txt_marca"),
            precio=request.POST.get("txt_precio"),
            tipo=request.POST.get("txt_tipo"),
            subtipo=request.POST.get("txt_subtipo"),
            categoria=request.POST.get("cmb_categoria"),
            stock=request.POST.get("txt_stock"),
            descripcion=request.POST.get("txa_descripcion"),
        )
    return redirect("/productosLista/")


@csrf_exempt
def editProducto(request, **kwargs):
    id = kwargs.get("id")
    producto = Producto.objects.get(id=id)
    if request.method == "POST":
        query = ""
        if "fil_foto" in request.FILES:
            fs = FileSystemStorage()
            archivo = request.FILES["fil_foto"]
            narchivo = fs.save(archivo.name, archivo)
            producto.foto = archivo.name
        producto.nombre = request.POST["txt_nombre"]
        producto.marca = request.POST["txt_marca"]
        producto.precio = request.POST["txt_precio"]
        producto.tipo = request.POST["txt_tipo"]
        producto.subtipo = request.POST["txt_subtipo"]
        producto.categoria = request.POST["cmb_categoria"]
        producto.stock = request.POST["txt_stock"]
        producto.descripcion = request.POST["txa_descripcion"]
        producto.save()
    return redirect("/productosLista/")

File: musicpro/static/js/productosLista.js
function modalNuevo() {
  $("#modalNuevo").modal('show');
}

function cargar(id) {
  $.post("../../producto/" + id + "/",{
      id:id
    },function(data){
      $("#modalModificar #txt_nombre").val(data[0]["nombre"]);
      $("#modalModificar #txt_tipo").val(data[0]["tipo"]);
      $("#modalModificar #txt_subtipo").val(data[0]["subtipo"]);
      $("#modalModificar #txt_marca").val(data[0]["marca"]);
      $("#modalModificar #txt_precio").val(data[0]["precio"]);
      $("#modalModificar #txt_stock").val(data[0]["stock"]);
      $("#modalModificar #img_producto").attr("src","/media/" + data[0]["foto"]);
      $("#modalModificar #cmb_categoria").val(data[0]["categoria"]);
      $("#modalModificar #txa_descripcion").val(data[0]["descripcion"]);
      $("#formModificar").attr("action","../../producto/" + id + "/editar/")
      $("#modalModificar").modal('show');
    });
}

$("#formNuevo").submit(function(e) {
  $("#modalNuevo #txt_precio").val($("#modalNuevo #txt_precio").val().replace('.',''));
  let nombre = $("#modalNuevo #txt_nombre").val(),
      tipo = $("#modalNuevo #txt_tipo").val(),
      subtipo = $("#modalNuevo #txt_subtipo").val(),
      marca = $("#modalNuevo #txt_marca").val(),
      precio = $("#modalNuevo #txt_precio").val(),
      stock = $("#modalNuevo #txt_stock").val(),
      descripcion = $("#modalNuevo #txa_descripcion").val(),
      foto = $("#modalNuevo #fil_foto").val();
  if (nombre.trim().length == 0 || tipo.trim().length == 0 || subtipo.trim().length == 0 || marca.trim().length == 0 || precio.trim().length == 0 || stock.trim().length == 0 || descripcion.trim().length == 0 || foto.trim().length == 0) {
    e.preventDefault();
    Swal.fire({
      icon: 'error',
      title: 'Oops...',
      text: "Favor ingrese todos los campos solicitados."
    });
    return false;
  }
  if ($.inArray(rol.trim(), ['Instrumentos de Cuerdas','Percusión','Amplificadores','Accesorios varios',]) == -1) {
    e.preventDefault();
    Swal.fire({
      icon: 'error',
      title: 'Oops...',
      text: "Ingrese una categoría válida."
    });
    return false;
  } 
});

$("#formModificar").submit(function(e) {
  $("#formModificar #txt_precio").val($("#formModificar #txt_precio").val().replace('.',''));
  let nombre = $("#modalModificar #txt_nombre").val(),
      tipo = $("#modalModificar #txt_tipo").val(),
      subtipo = $("#modalModificar #txt_subtipo").val(),
      marca = $("#modalModificar #txt_marca").val(),
      precio = $("#modalModificar #txt_precio").val(),
      stock = $("#modalModificar #txt_stock").val(),
      descripcion = $("#modalModificar #txa_descripcion").val();
  if (nombre.trim().length == 0 || tipo.trim().length == 0 || subtipo.trim().length == 0 || marca.trim().length == 0 || precio.trim().length == 0 || stock.trim().length == 0 || descripcion.trim().length == 0) {
    e.preventDefault();
    Swal.fire({
      icon: 'error',
      title: 'Oops...',
      text: "Favor ingrese todos los campos solicitados."
    });
    return false;
  }
  if ($.inArray(rol.trim(), ['Instrumentos de Cuerdas','Percusión','Amplificadores','Accesorios varios',]) == -1) {
    e.preventDefault();
    Swal.fire({
      icon: 'error',
      title: 'Oops...',
      text: "Ingrese una categoría válida."
    });
    return false;
  } 
});